Bug with Testing

- when running npm test, receive 2 errors that say: 
- Error in ./test/recipe-test.js
  Module not found: '../data/recipe-data' in '/Users/ashleymcdonald/turing/mod2projects/Whats-Cookin/test'

- Error in ./test/user-test.js
  Module not found: '../data/users-data' in '/Users/ashleymcdonald/turing/mod2projects/Whats-Cookin/test'

--- a/src/user.js
+++ b/src/user.js
@@ -24,8 +24,8 @@ class User {
     return this[recipeLocation].filter(recipe => recipe.type.includes(type));
   }
 
-  searchForRecipe(keyword) {
-    return this.favoriteRecipes.filter(recipe => recipe.name.includes(keyword) || recipe.ingredients.includes(keyword));
+  searchForRecipe(recipeLocation, keyword) {
+    return this[recipeLocation].filter(recipe => recipe.name.includes(keyword) || recipe.ingredients.includes(keyword));
   }
 }
 

--- a/src/user.js
+++ b/src/user.js
@@ -6,21 +6,24 @@ class User {
     this.favoriteRecipes = [];
     this.recipesToCook = [];
   }
+
   saveRecipe(recipe) {
     this.favoriteRecipes.push(recipe);
   }
 
-  removeRecipe(recipe) {
-    let i = this.favoriteRecipes.indexOf(recipe);
-    this.favoriteRecipes.splice(i, 1);
+  removeRecipe(recipeLocation, recipe) {
+    let recipeIndex = this[recipeLocation].indexOf(recipe);
+    this[recipeLocation].splice(recipeIndex, 1);
   }
 
   decideToCook(recipe) {
     this.recipesToCook.push(recipe);
   }
+
   filterRecipes(type) {
     return this.favoriteRecipes.filter(recipe => recipe.type.includes(type));
   }
+
   searchForRecipe(keyword) {
     return this.favoriteRecipes.filter(recipe => recipe.name.includes(keyword) || recipe.ingredients.includes(keyword));
   }

--- a/src/user.js
+++ b/src/user.js
@@ -20,8 +20,8 @@ class User {
     this.recipesToCook.push(recipe);
   }
 
-  filterRecipes(type) {
-    return this.favoriteRecipes.filter(recipe => recipe.type.includes(type));
+  filterRecipes(recipeLocation, type) {
+    return this[recipeLocation].filter(recipe => recipe.type.includes(type));
   }
 
   searchForRecipe(keyword) {

--- a/test/recipe-test.js
+++ b/test/recipe-test.js
@@ -1,7 +1,7 @@
 import { expect } from 'chai';
 
 import Recipe from '../src/recipe';
-import data from '../data/recipe-data';
+import data from '../src/data/recipe-data';
 
 describe('Recipe', function() {
   let recipe;

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -1,7 +1,7 @@
 import { expect } from 'chai';
 
 import User from '../src/user';
-import data from '../data/users-data';
+import data from '../src/data/users-data';
 
 describe('User', function() {
   let user;

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -51,10 +51,15 @@ describe('User', function() {
     expect(user.recipesToCook[0].name).to.equal('Chicken Parm');
   });
 
-  it('should be able to filter recipes by type', function() {
+  it('should be able to filter recipes by type in favorites', function() {
     user.saveRecipe(recipe);
     user.saveRecipe(recipe2);
-    expect(user.filterRecipes('italian')).to.deep.equal([recipe]);
+    expect(user.filterRecipes('favoriteRecipes', 'italian')).to.deep.equal([recipe]);
+  });
+
+  it('should be able to filter recipes by type in in recipes to cook', function() {
+    user.decideToCook(recipe2);
+    expect(user.filterRecipes('recipesToCook', 'american')).to.deep.equal([recipe2]);
   });
 
   it('should be able to search recipes by name', function() {

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -62,14 +62,24 @@ describe('User', function() {
     expect(user.filterRecipes('recipesToCook', 'american')).to.deep.equal([recipe2]);
   });
 
-  it('should be able to search recipes by name', function() {
+  it('should be able to search recipes by name in favorites', function() {
     user.saveRecipe(recipe);
-    expect(user.searchForRecipe('Chicken Parm')).to.deep.equal([recipe]);
+    expect(user.searchForRecipe('favoriteRecipes', 'Chicken Parm')).to.deep.equal([recipe]);
   });
 
-  it('should be able to search recipes by ingredient', function() {
+  it('should be able to search recipes by ingredient in favorites', function() {
     user.saveRecipe(recipe2);
-    expect(user.searchForRecipe('chicken wing')).to.deep.equal([recipe2]);
+    expect(user.searchForRecipe('favoriteRecipes', 'chicken wing')).to.deep.equal([recipe2]);
+  });
+
+  it('should be able to search recipes by name in recipes to cook', function() {
+    user.decideToCook(recipe);
+    expect(user.searchForRecipe('recipesToCook', 'Chicken Parm')).to.deep.equal([recipe]);
+  });
+
+  it('should be able to search recipes by ingredient in recipes to cook', function() {
+    user.decideToCook(recipe2);
+    expect(user.searchForRecipe('recipesToCook', 'chicken wing')).to.deep.equal([recipe2]);
   });
 
   it('should be able to remove recipe from favorites', () => {

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -9,7 +9,7 @@ describe('User', function() {
   let recipe;
 
   beforeEach(function() {
-    userInfo = data.users[0];
+    userInfo = data.filter(user => user.id === 1);
     user = new User(userInfo)
 
     recipe = {name: 'Chicken Parm', type: ['italian', 'dinner']};

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -3,13 +3,13 @@ import { expect } from 'chai';
 import User from '../src/user';
 import data from '../src/data/users-data';
 
-describe('User', function() {
+describe('User', () => {
   let user;
   let userInfo;
   let recipe;
   let recipe2;
 
-  beforeEach(function() {
+  beforeEach(() => {
     userInfo = data.filter(user => user.id === 1);
     user = new User(userInfo[0]);
 
@@ -17,67 +17,67 @@ describe('User', function() {
     recipe2 = { name: 'Buffalo Wings', type: ['american', 'lunch'], ingredients: ['chicken wing', 'red chili powder', 'tabasco sauce'] };
   });
 
-  it('should be a function', function() {
+  it('should be a function', () => {
     expect(User).to.be.a('function');
   });
 
-  it('should initialize with an id', function() {
+  it('should initialize with an id', () => {
     expect(user.id).to.equal(1);
   });
 
-  it('should initialize with a name', function() {
+  it('should initialize with a name', () => {
     expect(user.name).to.equal('Saige O\'Kon');
   });
 
-  it('should initialize with a pantry', function() {
+  it('should initialize with a pantry', () => {
     expect(user.pantry[0].ingredient).to.equal(11477);
   });
 
-  it('should initialize with an empty favoriteRecipes array', function() {
+  it('should initialize with an empty favoriteRecipes array', () => {
     expect(user.favoriteRecipes).to.deep.equal([]);
   });
 
-  it('should initialize with an empty recipesToCook array', function() {
+  it('should initialize with an empty recipesToCook array', () => {
     expect(user.recipesToCook).to.deep.equal([]);
   });
 
-  it('should be able to save a recipe to favoriteRecipes', function() {
+  it('should be able to save a recipe to favoriteRecipes', () => {
     user.saveRecipe(recipe);
     expect(user.favoriteRecipes[0].name).to.equal('Chicken Parm');
   });
 
-  it('should be able to decide to cook a recipe', function() {
+  it('should be able to decide to cook a recipe', () => {
     user.decideToCook(recipe);
     expect(user.recipesToCook[0].name).to.equal('Chicken Parm');
   });
 
-  it('should be able to filter recipes by type in favorites', function() {
+  it('should be able to filter recipes by type in favorites', () => {
     user.saveRecipe(recipe);
     user.saveRecipe(recipe2);
     expect(user.filterRecipes('favoriteRecipes', 'italian')).to.deep.equal([recipe]);
   });
 
-  it('should be able to filter recipes by type in recipes to cook', function() {
+  it('should be able to filter recipes by type in recipes to cook', () => {
     user.decideToCook(recipe2);
     expect(user.filterRecipes('recipesToCook', 'american')).to.deep.equal([recipe2]);
   });
 
-  it('should be able to search recipes by name in favorites', function() {
+  it('should be able to search recipes by name in favorites', () => {
     user.saveRecipe(recipe);
     expect(user.searchForRecipe('favoriteRecipes', 'Chicken Parm')).to.deep.equal([recipe]);
   });
 
-  it('should be able to search recipes by ingredient in favorites', function() {
+  it('should be able to search recipes by ingredient in favorites', () => {
     user.saveRecipe(recipe2);
     expect(user.searchForRecipe('favoriteRecipes', 'chicken wing')).to.deep.equal([recipe2]);
   });
 
-  it('should be able to search recipes by name in recipes to cook', function() {
+  it('should be able to search recipes by name in recipes to cook', () => {
     user.decideToCook(recipe);
     expect(user.searchForRecipe('recipesToCook', 'Chicken Parm')).to.deep.equal([recipe]);
   });
 
-  it('should be able to search recipes by ingredient in recipes to cook', function() {
+  it('should be able to search recipes by ingredient in recipes to cook', () => {
     user.decideToCook(recipe2);
     expect(user.searchForRecipe('recipesToCook', 'chicken wing')).to.deep.equal([recipe2]);
   });

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -10,7 +10,7 @@ describe('User', function() {
 
   beforeEach(function() {
     userInfo = data.filter(user => user.id === 1);
-    user = new User(userInfo)
+    user = new User(userInfo[0])
 
     recipe = {name: 'Chicken Parm', type: ['italian', 'dinner']};
   });
@@ -20,15 +20,15 @@ describe('User', function() {
   });
 
   it('should initialize with an id', function() {
-    expect(user.id).to.eq(1);
+    expect(user.id).to.equal(1);
   });
 
   it('should initialize with a name', function() {
-    expect(user.name).to.eq('Saige O\'Kon');
+    expect(user.name).to.equal('Saige O\'Kon');
   });
 
   it('should initialize with a pantry', function() {
-    expect(user.pantry[0].ingredient).to.eq(11477);
+    expect(user.pantry[0].ingredient).to.equal(11477);
   });
 
   it('should initialize with an empty favoriteRecipes array', function() {

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -61,4 +61,10 @@ describe('User', function() {
     user.saveRecipe(recipe);
     expect(user.searchForRecipe('Chicken Parm')).to.deep.equal([recipe]);
   });
+
+  it('should be able to remove recipe from favorites', () => {
+    user.saveRecipe(recipe);
+    user.saveRecipe(recipe2);
+    expect(user.removeRecipe(recipe)).to.deep.equal(user.favoriteRecipes[recipe2]);
+  });
 });

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -57,7 +57,7 @@ describe('User', function() {
     expect(user.filterRecipes('favoriteRecipes', 'italian')).to.deep.equal([recipe]);
   });
 
-  it('should be able to filter recipes by type in in recipes to cook', function() {
+  it('should be able to filter recipes by type in recipes to cook', function() {
     user.decideToCook(recipe2);
     expect(user.filterRecipes('recipesToCook', 'american')).to.deep.equal([recipe2]);
   });

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -65,6 +65,12 @@ describe('User', function() {
   it('should be able to remove recipe from favorites', () => {
     user.saveRecipe(recipe);
     user.saveRecipe(recipe2);
-    expect(user.removeRecipe(recipe)).to.deep.equal(user.favoriteRecipes[recipe2]);
+    expect(user.removeRecipe('favoriteRecipes', recipe)).to.deep.equal(user.favoriteRecipes[recipe2]);
+  });
+
+  it('should be able to remove recipe from recipes to cook', () => {
+    user.decideToCook(recipe);
+    user.decideToCook(recipe2);
+    expect(user.removeRecipe('recipesToCook', recipe2)).to.deep.equal(user.recipesToCook[recipe]);
   });
 });

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -7,12 +7,14 @@ describe('User', function() {
   let user;
   let userInfo;
   let recipe;
+  let recipe2;
 
   beforeEach(function() {
     userInfo = data.filter(user => user.id === 1);
     user = new User(userInfo[0])
 
     recipe = {name: 'Chicken Parm', type: ['italian', 'dinner']};
+    recipe2 = { name: 'Buffalo Wings', type: ['american', 'lunch'] };
   });
 
   it('should be a function', function() {
@@ -51,6 +53,7 @@ describe('User', function() {
 
   it('should be able to filter recipes by type', function() {
     user.saveRecipe(recipe);
+    user.saveRecipe(recipe2);
     expect(user.filterRecipes('italian')).to.deep.equal([recipe]);
   });
 

--- a/test/user-test.js
+++ b/test/user-test.js
@@ -11,10 +11,10 @@ describe('User', function() {
 
   beforeEach(function() {
     userInfo = data.filter(user => user.id === 1);
-    user = new User(userInfo[0])
+    user = new User(userInfo[0]);
 
-    recipe = {name: 'Chicken Parm', type: ['italian', 'dinner']};
-    recipe2 = { name: 'Buffalo Wings', type: ['american', 'lunch'] };
+    recipe = {name: 'Chicken Parm', type: ['italian', 'dinner'], ingredients: ['dry breadcrumbs', 'boneless skinless chicken breasts', 'parmesan cheese', 'marinara sauce']};
+    recipe2 = { name: 'Buffalo Wings', type: ['american', 'lunch'], ingredients: ['chicken wing', 'red chili powder', 'tabasco sauce'] };
   });
 
   it('should be a function', function() {
@@ -60,6 +60,11 @@ describe('User', function() {
   it('should be able to search recipes by name', function() {
     user.saveRecipe(recipe);
     expect(user.searchForRecipe('Chicken Parm')).to.deep.equal([recipe]);
+  });
+
+  it('should be able to search recipes by ingredient', function() {
+    user.saveRecipe(recipe2);
+    expect(user.searchForRecipe('chicken wing')).to.deep.equal([recipe2]);
   });
 
   it('should be able to remove recipe from favorites', () => {

